### DISCUSSION TITLE
Installation fails "error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly before end of the underlying stream"

### DISCUSSION THREAD

**Discussion by kieranmetcalfe** [Homebrew/discussions#3343] [Getting started] [answered]
Hi,

I can't get brew to install on a brand new mac (monterey).  I know this is similar to another post, but the error is different and the steps suggested there don't seem to work. 

I ran the installer, but it hung on installing command line tools, so I quit and downloaded that manually from Apple. 

On re-running the installer, it now crashes as follows:

==> /usr/bin/sudo /bin/mkdir -p /usr/local/Homebrew
==> /usr/bin/sudo /usr/sbin/chown -R kieran:admin /usr/local/Homebrew
==> Downloading and installing Homebrew...
remote: Enumerating objects: 209223, done.
remote: Counting objects: 100% (95/95), done.
remote: Compressing objects: 100% (79/79), done.
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly before end of the underlying stream
error: 2227 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Failed during: git fetch --force origin

Any ideas how to solve this? I need to use brew to install PHP...

**Comment by kieranmetcalfe** [comment, marked as answer]
Sorry - ignore this. Looks like it might have been clashing with a Dropbox file transfer. Very odd.

**Comment by sleekLancelot** [comment]
I was stuck on this for hours, all I did was restart my system and it worked.

**Comment by pelf** [comment]
I had a similar issue. Rebooting didn't help. It seems to have been caused by the system downloading an OS upgrade at the same time. Maybe that was starving the homebrew download of bandwidth? Once the upgrade was finished, homebrew installed fine.

**Comment by Dzannypy** [comment]
solution i found is to run `git config --global http.version HTTP/1.1` then try installing again

**Reply by miroaccgh** [reply]
awesome thanks

**Reply by Henymoon** [reply]
thank you!

**Reply by tedlandis** [reply]
That did it!

**Reply by pfaaworkin01** [reply]
Worked for me as well, thanks brother!!!

**Reply by yubrajniraula** [reply]
worked like a charm. Thank you

**Reply by rgjacobx2** [reply]
This helped. Thanks.

**Reply by Ntochukwu-David** [reply]
Thanks this command works fine.

**Reply by mrg123** [reply]
my new Mac mini M4, failed by this issue , can't help me out

**Reply by ensolaris** [reply]
New Mac M4, helped. Thanks!

**Reply by navi504** [reply]
thanks, solved my problem on m4 pro

**Reply by ssi-anik** [reply]
Others are saying that this one helped. But in my case, it didn't. But the following link did. Might help others.

https://dev.to/styt/solving-error-rpc-failed-curl-92-http2-stream-5-was-not-closed-cleanly-cancel-err-8-during-homebrew-installation-4f0f

**Reply by pjincz** [reply]
You saved my day, bro.

**Reply by nspady** [reply]
Thanks for this, same issue on my M4 MBP.

**Reply by codeinearts** [reply]
Same here with Mac M4, this did the trick. Thanks!

**Reply by gaurav23122001** [reply]
Hi, my installation is stuck here on M4 PRO for some reason. What to do. Previously I was getting the error but that is gone after running the above command.
<img width="677" alt="Screenshot 2025-05-06 at 9 39 33 PM" src="https://github.com/user-attachments/assets/d236b804-8dce-496f-9a50-5a2b12311f78" />
